warehouse/views.py

```diff
@@ -399,14 +399,16 @@ def stats(request):
     top_100_packages = (
         request.db.query(Project)
         .with_entities(Project.name, Project.total_size)
+        .filter(Project.total_size > 0)
         .order_by(Project.total_size.desc())
         .limit(100)
         .all()
     )
     # Move top packages into a dict to make JSON more self describing
     top_packages = {
-        pkg_name: {"size": int(pkg_bytes) if pkg_bytes is not None else 0}
+        pkg_name: {"size": int(pkg_bytes)}
         for pkg_name, pkg_bytes in top_100_packages
+        if pkg_bytes is not None
```

Comment:
This conditional would still have the effect of removing some packages from the 100 found in the query, no?

I think the original issue is that `total_size` is null, not that it's zero. The reason it's appearing at the top of the results is because our ORM orders nulls before non-null values. The reason it's appearing as zero is because of the `if pkg_bytes is not None else 0` in the comprehension.

I think instead we can just tell it to put nulls last when ordering:

```python
.order_by(Project.total_size.desc().nullslast())
```

Reply:
No problem, I'll add it - but shouldn't this conditional remain in case null sizes still make it to the top 100 even with reordering? or it's too unlikely....

Reply:
I think it's pretty unlikely, but the original `if pkg_bytes is not None else 0` should handle that if it happens, so that should probably remain.

Reply:
Ok thanks - and from behavioural POV, in case of None do we want to have it displayed and keep the cap at 100 or do we prefer to filter it out and have potentially less then 100 (in the unlikely event)?

Reply:
We want the current behavior (displayed as zero) we just want it to come last.

Reply:
Ok, so essentially I've simply added the _.nullslast()_ like you've suggested and that produces the behaviour needed - nulls will only get pulled from db in an unlikely event and in that case get assigned a zero value and appear last on the table.